src/monster.h

```diff
@@ -164,6 +165,9 @@ class monster : public Creature, public JsonSerializer, public JsonDeserializer
         int calc_movecost( const tripoint &f, const tripoint &t ) const;
         int calc_climb_cost( const tripoint &f, const tripoint &t ) const;
 
+        bool is_dangerous_field( const field &fld ) const;
+        bool is_dangerous_field( const field_entry &entry ) const;
```

Comment:
`Character` already has `is_dangerous_field()`s. It would be much better to make them `virtual` and `override` some of them in `monster`.